base/show.jl

```diff
@@ -883,99 +883,115 @@ function show(io::IO, tv::TypeVar)
     end
 end
 
-# dump & xdump - structured tree representation like R's str()
-# - dump is for the user-facing structure
-# - xdump is for the internal structure
-#
-# x is the object
-# n is the depth of traversal in nested types (5 is the default)
-# indent is a character string of spaces that is incremented at
-# each descent.
-#
-# Package writers may overload dump for other nested types like lists
-# or DataFrames. If overloaded, check the nesting level (n), and if
-# n > 0, dump each component. Limit to the first 10 entries. When
-# dumping components, decrement n, and add two spaces to indent.
-#
-# Package writers should not overload xdump.
+function dump(io::IO, x::SimpleVector, n::Int, indent)
+    if isempty(x)
+        print(io, "empty SimpleVector")
+        return
+    end
+    print(io, "SimpleVector")
+    if n > 0
+        for i = 1:length(x)
+            println(io)
+            print(io, indent, "  ", i, ": ")
+            if isassigned(x,i)
+                dump(io, x[i], n - 1, string(indent, "  "))
+            else
+                print(io, undef_ref_str)
+            end
+        end
+    end
+end
 
-function xdump(fn::Function, io::IO, x, n::Int, indent)
+function dump(io::IO, x::ANY, n::Int, indent)
     T = typeof(x)
-    print(io, T, " ")
-    if isa(T, DataType) && nfields(T) > 0
-        println(io)
+    if isa(x, Function)
+        print(io, x, " (function of type ", T, ")")
+    else
+        print(io, T)
+    end
+    if nfields(T) > 0
         if n > 0
-            for field in fieldnames(T)
-                if field != symbol("")    # prevents segfault if symbol is blank
-                    print(io, indent, "  ", field, ": ")
-                    if isdefined(x,field)
-                        fn(io, getfield(x, field), n - 1, string(indent, "  "))
-                    else
-                        println(io, undef_ref_str)
-                    end
+            for field in (isa(x,Tuple) ? (1:length(x)) : fieldnames(T))
```

Comment:
should probably always be `1:nfields(T)`, since the indices are unique and always available